scripts/init.mk

```diff
@@ -49,11 +49,20 @@ githooks-run: # Run git hooks configured in this repository @Operations
 		--all-files
 
 asdf-install: # Install asdf @Installation
-	if [ -d "${HOME}/.asdf" ]; then
-		( cd "${HOME}/.asdf"; git pull )
-	else
-		git clone --depth=1 https://github.com/asdf-vm/asdf.git "${HOME}/.asdf" ||:
+	asdf_version=$$(grep -E "#\s+asdf\s+v[0-9.]+\s+[a-fA-F0-9]{40}" .tool-versions | awk '{ print $$4 }')
+	if ! [ -d "${HOME}/.asdf" ] || ! [ -d "${HOME}/.asdf/.git" ]; then
+		rm -rf "${HOME}/.asdf"
+		git clone https://github.com/asdf-vm/asdf.git "${HOME}/.asdf" ||:
 	fi
+	cd "${HOME}/.asdf"
+	git checkout master
+	git pull
+	git checkout $$asdf_version
```

Comment:
This means:

1) We're taking responsibility for the specific asdf version on the developer's machine, over and above any install they may already have; that means updating all repositories in response to upstream asdf bugfixes and security problems
2) We need to be sure we can keep the asdf version locked across all `repository-template` forks, otherwise two projects specifying different versions will end up with one of them being wrong.

This is where it starts to feel like we're trying to do too much in this repo.  Can we have a principle that make tasks for a project should only be scoped to that project's directory?  If we want to arrange standard tooling to be installed on a dev machine, it feels like the best place we've currently got for that is `dotfiles`.

Reply:
Agree  - also trying to control the version of what is a tool version manager seems quite meta.

Reply:
In my opinion, the whole idea of `asdf` is to help resolve these kinds of issues: scoping runtime dependencies down to individual projects and repositories, and avoiding side-effects. I'd rather we recommend that the latest version of `asdf` is installed on a developers environment (and offer an easy mechanism to do so), and always default to using the latest version within CI/CD pipelines.

Always using the latest version of `asdf` shouldn't be a problem (apart from breaking CLI changes, which are rare). The things that matter are the runtime dependencies used for building, testing and deploying the application. If someone is running an old version of `asdf` and the runtime versions are not available, that should be a prompt for them to update their version of `asdf`.

Reply:
Thanks all for your valuable comments and online/offline discussions.

We agreed that `asdf` is the right tool for the job. However, as multiple considerations have to be taken into account, e.g. how this tool is installed, OS-level dependencies, the version in use, etc., we concluded that it is outside the remit of the repository template to make these recommendations/decisions. Therefore, the expectation is that the engineering team responsible for maintaining their repository will make these choices.